### ISSUE TITLE
Upon error code 2 from acme.sh, certs.sh does not correctly trap and destroys the TLS secrets

### ISSUE THREAD

**Issue by coresci-sharper** [math-nao/certs#28]
Reviewing https://github.com/math-nao/certs/blob/master/scripts/certs.sh ....

Line 311 is a return on when acme.sh with return code 2 (no update needed) is fired. This appears not to be honored as acme.sh returns an error code of 2 (no update required) and the _add_certs_to_secret_ function is executed anyway.  In a loop with multiple domains and multiple ingress definitions, this results in two issues:

- If the cert is the first cert to be renewed, the secret is blown out with null values as the cat command finds no file
- If a previous execution loop was successful, the previous loop's TLS files are leveraged for the wrong domain/loop because of a name collision in the hardcoded pem filenames.

Example of the first state:
```
Info: Loading conf from secret...
Info: Adding conf
[Wed Jul 29 12:00:37 UTC 2020] Domains not changed.
[Wed Jul 29 12:00:37 UTC 2020] Skip, Next renewal time is: Mon Sep 21 23:55:12 UTC 2020
[Wed Jul 29 12:00:37 UTC 2020] Add '--force' to force to renew.
Info: acme.sh return code: 2
Info: Certificate change, updating...
Info: Adding certs to secret...
cat: /root/certs/ca.crt: No such file or directory
cat: /root/certs/fullchain.crt: No such file or directory
cat: /root/certs/tls.key: No such file or directory
Info: Updating certs
Info: Certs sucessfully added
Info: Adding conf to secret...
Info: domain folder is not a directory
Info: No cmd args found
```

Inspecting the secret:
```
$ kubectl get secrets -n **** helm-****************-tls-secrets -o yaml
apiVersion: v1
data:
  ca.crt: ""
  tls.crt: ""
  tls.key: ""
kind: Secret
metadata:
  creationTimestamp: "2020-07-23T23:55:12Z"
  name: helm-****************-tls-secrets
  namespace: *****
  resourceVersion: "2*******3"
  selfLink: /api/v1/namespaces/*****/secrets/helm-****************-tls-secrets
  uid: 00000000-0000-0000-0000-000000000000
type: Opaque
```

In the second state... you can easily see this by inspecting the certificate after certs.sh runs acme.sh looping. Simply output the secret and push it into openssl and inspect the CN in the cert:

```
$ kubectl get secrets -n ***** helm-****************-tls-secrets -o 'go-template={{index .data "tls.crt"}}' | base64 -D | openssl x509 -noout -text
```

We've had this result in bad certificates (the default kubernetes ingress certificate because tls.crt and tls.key are null) and wrong certificates (domain 2 has domain 1's cert) in production environments... which is bad.

This also can lead to a situation where multiple runs to "jiggle the handle" to fix stuff hits the LetsEncrypt API restrictions.  This is also bad when you cannot generate a cert for your production environment because acme.sh has reached out for a new cert 5 times in a week.

**Comment by borigas** [comment]
What version are you on? This sounds a lot like a bug I fixed in #25 that's available in version 1.1.6

**Comment by math-nao** [comment]
@coresci-sharper could you please enable `debug` mode [doc](https://github.com/math-nao/certs#chart-configuration) and post your logs?
